src/Sulu/Bundle/SnippetBundle/Admin/SnippetAdmin.php

```diff
@@ -100,22 +100,30 @@ function(Localization $localization) {
                 ->addOption('title', 'sulu_snippet.snippets')
                 ->addOption('resourceKey', 'snippets')
                 ->addOption('adapters', ['table'])
-                ->addOption('editRoute', 'sulu_snippet.form.detail')
+                ->addOption('addRoute', 'sulu_snippet.add_form.detail')
+                ->addOption('editRoute', 'sulu_snippet.edit_form.detail')
                 ->addOption('locales', $snippetLocales)
                 ->addAttributeDefault('locale', $snippetLocales[0]),
-            (new Route('sulu_snippet.form', '/snippets/:locale/:id', 'sulu_admin.resource_tabs'))
+            (new Route('sulu_snippet.add_form', '/snippets/:locale/add', 'sulu_admin.resource_tabs'))
                 ->addOption('resourceKey', 'snippets')
                 ->addOption('locales', $snippetLocales),
-            (new Route('sulu_snippet.form.detail', '/details', 'sulu_admin.form'))
+            (new Route('sulu_snippet.add_form.detail', '/details', 'sulu_admin.form'))
```

Comment:
The `sulu_snippet.add_form.detail` and `sulu_snippet.edit_form.detail` are almost identical. The only difference is the parent route. Maybe we should think about a solution for that. I came up with two different solutions:

1. Create a `withParent` method on the route. This works similar like for PSR-7 requests, the method would return a cloned route with this parent set. Then we could do something like this:

```php
$formDetailRoute = new Route(...);

return [
    $formDetailRoute->withParent('sulu_snippet.add_form'),
    $formDetailRoute->withParent('sulu_snippet.edit_form'),
];
```

2. Allow a route to have multiple parents

Reply:
not yet